app/models/search_builder.rb

```diff
@@ -18,6 +22,13 @@ def join_from_parent(solr_params)
     solr_params[:q] = JoinChildrenQuery.new(solr_params[:q]).to_s
   end
 
+  def hide_incomplete(solr_params)
+    return if current_user.admin? || current_user.editor? || current_user.curator? || current_user.fulfiller? || current_user.image_editor?
+    solr_params[:fq] ||= []
+    state_field = ActiveFedora.index_field_mapper.solr_name('state', :symbol)
+    solr_params[:fq] << "#{state_field}:complete OR #{state_field}:flagged"
```

Comment:
I really wanted to use Ability to do this bit, but it requires a record to check against which has a persisted ID (because of Blacklight::AccessControls). I could move this method over there, but I don't think `ability.rb` would use it, since it uses CanCan's DSL.